tests/petsc_complex/vector_02.cc

```diff
@@ -0,0 +1,113 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2004 - 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+// check assignment of elements in Vector
+
+#include "../tests.h"
+#include <deal.II/lac/petsc_vector.h>
+#include <deal.II/lac/vector.h>
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+
+void test ()
+{
+  const unsigned int s = 10;
+  PETScWrappers::Vector v(s);
+  for (unsigned int k=0; k<v.size(); ++k)
+    v(k) = k;
+
+  v.compress (VectorOperation::insert);
+
+  PETScWrappers::Vector v2(s);
+  for (unsigned int k=0; k<v2.size(); ++k)
+    {
+      // This fails
+      // v2(k) = PetscScalar (k,-k);
```

Comment:
you were right, it had to do with `unsigned int`. I suppose `-k` led to something like max unsigned int.